code/go/internal/validator/semantic/version_integrity.go

```diff
@@ -127,3 +154,43 @@ func ensureManifestVersionHasChangelogEntry(manifestVersion string, versions []s
 	}
 	return errors.New("current manifest version doesn't have changelog entry")
 }
+
+func ensureLinksAreValid(links []string) error {
+
+	type validateFn func(link *url.URL) error
+
+	validateLinks := []struct {
+		domain       string
+		validateLink validateFn
+	}{
+		{
+			"github.com",
+			validateGithub,
+		},
+	}
+	for _, link := range links {
```

Comment:
This is the reason why we have `ValidationErrors` (plural). You can run the validation against all entries and combine errors into an array.

Reply:
The individual errors are now being appended to `ValidationErrors`.